### DISCUSSION TITLE
CPU usage settings

### DISCUSSION THREAD

**Discussion by jonasengelmann** [oxigraph/oxigraph#789] [General]
I am considering switching from blazegraph to oxigraph, however I did run into some performance issues when testing. In my tests blazegraph outperformed oxigraph on almost every query. Yet my tests were a bit unfair, as I noticed oxigraph would at most only use one CPU core whereas blazegraph made use of all available 8. How do I allow oxigraph to use all cores? I did run these tests in docker, but I don't think docker is the issue here. I even tried setting "cpus: 8.0" in my docker-compose, but without any effect. Are there some other settings I can tweak, sadly I could not find any config file. Or do you have other suggestions?

**Comment by Tpt** [comment]
> In my tests blazegraph outperformed oxigraph on almost every query

Yes, sadly Oxigraph is still way slower than Blazegraph on non-trivial queries. Query optimization is still basically non existent in Oxigraph. This is an expected result.

>  I noticed oxigraph would at most only use one CPU core whereas blazegraph made use of all available 8.

Oxigraph query execution is single threaded. To make use all your cores you need to fire queries in parallel.

> Or do you have other suggestions?

None outside of helping us improving Oxigraph ;). I focused most of my time of getting the basics right (parsers...) and did not had time yet for query planning/execution optimizations.
